server/message_hooks_test.go

```diff
@@ -1995,3 +1995,185 @@ func TestGetChatIDForChannel(t *testing.T) {
+			SetupAPI:        func(api *plugintest.API) {},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChat", testutils.GetChatID()).Return(&msteams.Chat{}, nil)
+			},
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 0,
+		},
+		{
+			Name:            "GetMentionsData: mentioned all in a channel",
+			Message:         "Hi @all",
+			ExpectedMessage: "Hi <at id=\"0\">Everyone</at>",
+			ChatID:          testutils.GetChatID(),
+			SetupAPI:        func(api *plugintest.API) {},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChat", testutils.GetChatID()).Return(&msteams.Chat{
+					Type: "G",
+				}, nil)
+			},
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: error occurred while getting chat",
+			Message:         "Hi @all",
+			ExpectedMessage: "Hi <at id=\"0\">@all</at>",
+			ChatID:          testutils.GetChatID(),
+			SetupAPI: func(api *plugintest.API) {
+				api.On("LogDebug", "Unable to get ms teams chat", "Error", mock.Anything)
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChat", testutils.GetChatID()).Return(nil, errors.New("error occurred while getting chat"))
+			},
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: mentioned all but chat ID is missing",
+			Message:         "Hi @all",
+			ExpectedMessage: "Hi <at id=\"0\"></at>",
+			SetupAPI:        func(api *plugintest.API) {},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChannelInTeam", testutils.GetTeamID(), testutils.GetChannelID()).Return(&msteams.Channel{}, nil)
+			},
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: error occurred while getting channel in team",
+			Message:         "Hi @all",
+			ExpectedMessage: "Hi <at id=\"0\">@all</at>",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("LogDebug", "Unable to get ms teams channel", "Error", mock.Anything)
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChannelInTeam", testutils.GetTeamID(), testutils.GetChannelID()).Return(nil, errors.New("error occurred while getting channel in team"))
+			},
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: mentioned a user",
+			Message:         "Hi @test-username",
+			ExpectedMessage: "Hi <at id=\"0\"></at>",
+			ChatID:          testutils.GetChatID(),
+			SetupAPI: func(api *plugintest.API) {
+				api.On("GetUserByUsername", "test-username").Return(&model.User{
+					Id: testutils.GetID(),
+				}, nil)
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetUser", testutils.GetUserID()).Return(&msteams.User{}, nil)
```

Comment:
i think we should fill in this data, as the expected message will change on these values. fix everyhwere